core/org.eclipse.cdt.core/utils/org/eclipse/cdt/utils/elf/Elf.java

```diff
@@ -340,7 +339,9 @@ public class Section {
 		public ByteBuffer mapSectionData() throws IOException {
 			makeSureNotCompressed();
 			areSectionsMapped = true;
-			return efile.getChannel().map(MapMode.READ_ONLY, sh_offset, sh_size).load().asReadOnlyBuffer();
+			ByteBuffer buff = ByteBuffer.allocateDirect((int) sh_size);
```

Comment:
If this is needed, I think that you shouldn't be changing this method, instead have the caller call loadSectionData, or possibly create a new method that returns a ByteBuffer.

Ideally the same fix should be applied to the other readers that have the same issue, e.g. Coff/Coff64

Reply:
I am trying to understand why do we need new method.. current and previous approach both uses `DirectByteBuffer` so ideally there should be no issue..

I'll checked other readers as well